src/ens.sol

```diff
@@ -0,0 +1,50 @@
+interface IReverseRegistrar {
+    function claim(address owner) external returns (bytes32 node);
+    function claimWithResolver(address owner, address resolver) external returns (bytes32 node);
+    function setName(string calldata name) external returns (bytes32 node);
+    function node(address) external view returns (bytes32 node);
+}
+
+interface INameResolver {
+    function name(bytes32 node) external view returns (string memory);
+}
+
+interface IENS {
+    function setSubnodeRecord(bytes32 node, bytes32 label, address owner, address resolver, uint64 ttl) external;
+    function resolver(bytes32 node) external view returns (address);
+    function owner(bytes32 node) external view returns (address);
+    function setResolver(bytes32 node, address resolver) external;
+}
+
+interface IAddrResolver {
+    function addr(bytes32 node) external view returns (address);
+}
+
+IReverseRegistrar constant REVERSE_REGISTRAR = IReverseRegistrar(0xa58E81fe9b61B5c3fE2AFD33CF304c454AbFc7Cb);
+IENS constant ENS = IENS(0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e);
+bytes32 constant ADDR_REVERSE_NODE = 0x91d1777781884d03a6757a803996e38de2a42967fb37eeaca72729271025a9e2;
+
+bytes32 constant sha3HexLookup = 0x3031323334353637383961626364656600000000000000000000000000000000;
```

Comment:
Explain please the special values

Reply:
its taken from the ens code that we copied over, its basically `"0123456789abcdef"` as bytes32, changed it to that